### PR TITLE
Akka cluster streamlet example

### PR DESCRIPTION
**Pull request by nolangrace**
### What changes were proposed in this pull request?

New example has been added to demonstrate the capabilities of Clustering a Cloudflow Akka Streamlet

### Why are the changes needed?

Help users understand how to use Clustering with a Cloudflow Akka Streamlet

### How was this patch tested?

Manually tested on GKE and runLocal

### CODE REVIEW

**Comment by nolangrace** [comment]
This example depends on Pull Request #234 being merged to master

File: examples/connected-car-cluster-sharding/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+$ kubectl cloudflow help
+This command line tool can be used to deploy and operate Cloudflow applications.
+...
+```
+
+* Deploy the app.
+
+```
+$ kubectl cloudflow deploy -u oauth2accesstoken eu.gcr.io/my-awesome-project/connected-car-akka-cluster:221-65c1693-dirty -p $(gcloud auth print-access-token)"
+Existing value will be used for configuration parameter 'cdr-generator2.records-per-second'
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/README.md:78]: Is this output from call record aggregator by mistake?

File: examples/connected-car-cluster-sharding/akka-connected-car-streamlet/src/main/scala/connectedcar/streamlets/RawCarDataIngress.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import cloudflow.akkastream.AkkaStreamlet
+import cloudflow.akkastream.scaladsl.RunnableGraphStreamletLogic
+import cloudflow.streamlets.StreamletShape
+import cloudflow.streamlets.avro.AvroOutlet
+import connectedcar.data.ConnectedCarERecord
+
+import scala.util.Random
+import scala.concurrent.duration._
+
+object RawCarDataIngress extends AkkaStreamlet {
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/akka-connected-car-streamlet/src/main/scala/connectedcar/streamlets/RawCarDataIngress.scala:14]: Please rename to RawCarDataGenerator, so it is clear that this just generates some data.

File: examples/connected-car-cluster-sharding/README.md (hunk 1 above)
Comment (nolangrace) [reply examples/connected-car-cluster-sharding/README.md]: Good catch!  sorry forgot to replace this section

File: examples/connected-car-cluster-sharding/akka-connected-car-streamlet/src/main/scala/connectedcar/streamlets/RawCarDataIngress.scala (hunk 2 above)
Comment (nolangrace) [reply examples/connected-car-cluster-sharding/akka-connected-car-streamlet/src/main/scala/connectedcar/streamlets/RawCarDataIngress.scala]: done

File: examples/connected-car-cluster-sharding/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
+## Akka Cluster Sharding Cloudflow Application
+
+### Problem Definition
+
+This project clusters a set of Akka Streamlets to demonstrate how leverage
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/README.md:5]:
```suggestion
This project clusters a set of Akka Streamlets to demonstrate how to leverage
```

File: examples/connected-car-cluster-sharding/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+Akka Cluster Sharding for stateful stream processing in Cloudflow
+
+![](akka-cluster-streams.png)
+
+### Sub projects
+
+This application consists of the following sub-projects:
+
+* `akka-connected-car`: Contains the blueprint
+* `akka-connected-car-streamlet`: Akka Streams based ingress, cluster streamlet, and logger
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/README.md:15]: ingress -> generator?

File: examples/connected-car-cluster-sharding/README.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+* `datamodel`: Contains the Avro schema
+
+### Example Deployment example on GKE
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/README.md:18]: BTW have you verified that with runLocal, everything also works as expected?

File: examples/connected-car-cluster-sharding/build.sbt [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import sbt._
+import sbt.Keys._
+
+val AkkaVersion         = "2.5.29"
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/build.sbt:4]: super minor, strange formatting

File: examples/connected-car-cluster-sharding/build.sbt [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    )
+
+lazy val connectedCarExample = (project in file("./akka-connected-car"))
+  .enablePlugins(CloudflowApplicationPlugin)
+  .settings(
+    commonSettings,
+    name := "connected-car-akka-cluster",
+    libraryDependencies ++= Seq(
+      "ch.qos.logback" %  "logback-classic" % "1.2.3",
+      "org.scalatest"          %% "scalatest"              % "3.0.7"    % "test"
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/build.sbt:29]: minor, can improve formatting

File: examples/connected-car-cluster-sharding/build.sbt [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  .enablePlugins(CloudflowLibraryPlugin)
+
+lazy val akkaConnectedCar= (project in file("./akka-connected-car-streamlet"))
+  .enablePlugins(CloudflowAkkaStreamsLibraryPlugin)
+  .settings(
+    commonSettings,
+    name := "akka-connected-car-streamlet",
+    libraryDependencies ++= Seq(
+      "com.typesafe.akka" %% "akka-cluster-sharding" % AkkaVersion,
+      "net.liftweb" %% "lift-json" % "3.3.0"
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/build.sbt:44]: Is this used?

File: examples/connected-car-cluster-sharding/README.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+Akka Cluster Sharding for stateful stream processing in Cloudflow
+
+![](akka-cluster-streams.png)
+
+### Sub projects
+
+This application consists of the following sub-projects:
+
+* `akka-connected-car`: Contains the blueprint
+* `akka-connected-car-streamlet`: Akka Streams based ingress, cluster streamlet, and logger
```
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/README.md:15]:
```suggestion
* `akka-connected-car-streamlet`: Akka Streams based generator, cluster streamlet, and logger
```

File: examples/connected-car-cluster-sharding/build.sbt (hunk 6 above)
Comment (RayRoestenburg) [inline examples/connected-car-cluster-sharding/build.sbt:4]:
```suggestion
val AkkaVersion = "2.5.29"
```

File: examples/connected-car-cluster-sharding/build.sbt (hunk 8 above)
Comment (RayRoestenburg) [reply examples/connected-car-cluster-sharding/build.sbt:44]: I'll try it locally and see.

File: examples/connected-car-cluster-sharding/README.md (hunk 5 above)
Comment (RayRoestenburg) [reply examples/connected-car-cluster-sharding/README.md:18]: Stupid question, of course you did.. I also checked and it works nicely

File: examples/connected-car-cluster-sharding/build.sbt (hunk 8 above)
Comment (RayRoestenburg) [reply examples/connected-car-cluster-sharding/build.sbt:44]: does not seem to be necessary, I'll remove it.

Comment (RayRoestenburg) [reply examples/connected-car-cluster-sharding/build.sbt:44]: I'll just re-publish this on another branch and merge it.

**Comment by RayRoestenburg** [comment]
@nolangrace I'll just make the little changes I found in here https://github.com/lightbend/cloudflow/pull/242 and merge it.

**Comment by RayRoestenburg** [comment]
This is Merged in another PR https://github.com/lightbend/cloudflow/pull/242

**Pull request closed without merging**